lib/assembly_report/assembly_report.hpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2014-2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ASSEMBLY_REPORT_HPP
+#define ASSEMBLY_REPORT_HPP
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string/trim.hpp>
+#include "util/stream_utils.hpp"
+#include "util/string_utils.hpp"
+
+namespace ebi
+{
+  namespace assembly_report
+  {
+    /*
+     * Struct to contain all the synonyms of a contig
+     */
+    struct ContigSynonyms {
+        std::vector<std::string> synonyms;
+
+        ContigSynonyms() {
+
+        }
+
+        ContigSynonyms(std::string contig) {
+            synonyms.push_back(contig);
+        }
+
+        ContigSynonyms(std::vector<std::string> & synonyms) : synonyms(synonyms) {
+
+        }
+    };
+
+    /*
+     * Map to map all contigs to their synonym_maps
+     */
+    class Synonyms_map {
```

Comment:
for classes we use UpperCamelCase. SynonymsMap is ok. SynonymsMapping is also ok.